arangod/StorageEngine/TransactionState.cpp

```diff
@@ -54,7 +56,16 @@ TransactionState::TransactionState(TRI_vocbase_t& vocbase,
       _hints(),
       _options(options),
       _serverRole(ServerState::instance()->getRole()),
-      _registeredTransaction(false) {}
+      _registeredTransaction(false) {
+  if (isCoordinator()) {
```

Comment:
we have to move this to `ClusterTransactionState::beginTransaction()`, of course we don't need `isCoordinator()` check

Reply:
Good point about ClusterTransactionState. But beginTransaction is to late. Still need to do it in ctor, as optimization step need to know revision.